### DISCUSSION TITLE
Is there a way to know the cost to send UDP data from one node to another?

### DISCUSSION THREAD

**Discussion by rprj** [openthread/openthread#9290] [Q&A] [answered]
Hi there,

If I want to send a UDP data from one node device to another in the Thread network, is there a way to know what the cost was and how many hops it took to reach that node? Thank you!

**Comment by abtink** [comment, marked as answer]
`otThreadGetNextHopAndPathCost()` API can be used:
https://github.com/openthread/openthread/blob/ce4462535773d418066de78e61ef1197c3cff425/include/openthread/thread_ftd.h#L896-L912

It is also available under CLI command `nexthop`
- [Code](https://github.com/openthread/openthread/blob/ce4462535773d418066de78e61ef1197c3cff425/src/cli/cli.cpp#L4841-L4937)
- Docs [nexthop_get](https://openthread.io/reference/cli/commands#nexthop_get) &  [nexthop](https://openthread.io/reference/cli/commands#nexthop)

**Reply by rprj** [reply]
Thank you!
